ksqldb-serde/src/main/java/io/confluent/ksql/serde/avro/AvroSRSchemaDataTranslator.java

```diff
@@ -53,22 +149,12 @@ public Object toConnectRow(final Object ksqlData) {
       );
       originalData = ConnectSchemas.withCompatibleRowSchema(originalData, originalSchema);
     }
+
     validate(originalSchema, schema);
```

Comment:
Yeah, just checked the validation code doesn't check the fields recursively!

Reply:
As discussed, we wont be implementing validation recursively, since we decided to keep it only for the root fields. Avro schemas are validated anyways recursively in the copy method